internal/jobqueue/dbjobqueue/dbjobqueue.go

```diff
@@ -246,6 +322,21 @@ func (q *DBJobQueue) Dequeue(ctx context.Context, jobTypes []string, channels []
 
 	return id, token, dependencies, jobType, args, nil
 }
+
+// dequeueMaybe is just a smaller helper for acquiring a connection and
+// running the sqlDequeue query
+func (q *DBJobQueue) dequeueMaybe(ctx context.Context, token uuid.UUID, jobTypes []string, channels []string) (id uuid.UUID, jobType string, args json.RawMessage, err error) {
+	var conn *pgxpool.Conn
+	conn, err = q.pool.Acquire(ctx)
+	if err != nil {
+		return
+	}
+	defer conn.Release()
+
+	err = conn.QueryRow(ctx, sqlDequeue, token, jobTypes, channels).Scan(&id, &jobType, &args)
```

Comment:
Should we not handle the error?

Reply:
`err` is returned via named return values.

Reply:
fancy!